### ISSUE TITLE
CBZ extracting error

### ISSUE THREAD

**Issue by m0lDaViA** [mienaiyami/yomikiru#312]
### Type of installation

Setup (.exe)

### Type of reader (if reader related)

image

### Steps to reproduce

Try to read a cbz.

### ✔️ Expected Behavior

Extracting should work normally and you should be able to see the image.

### ❌ Actual Behavior

Getting an error as shown in the picture.
![yomikiru](https://github.com/mienaiyami/yomikiru/assets/37399229/addd9db8-d372-42d2-94cb-7ebbf6c06e86)

**Comment by m0lDaViA** [comment]
I did and it worked totally fine.

**Comment by Skitch90** [comment]
> I did and it worked totally fine.

Did you have ConstrainedLanguage mode on?

To check it open powershell and run the following command;
`$ExecutionContext.SessionState.LanguageMode`

**Comment by m0lDaViA** [comment]
It says "ConstrainedLanguage". What does that mean?

**Comment by Skitch90** [comment]
> It says "ConstrainedLanguage". What does that mean?

Basically it's a mode that limits a lot powershell capabilities. Capabilities that can be exploited from malware. I don't know if there is a way to perform archive extraction with this mode.
If you put it yourself you could think to change the language mode back. In my case I decided to leve the ConstrainedLanguage mode, so I installed the application using WSL and run the application using WSLg

**Comment by mienaiyami** [comment]
The extraction library I use uses PowerShell(windows) and unzip(linux) behind it. I selected it at that time because I didnt want to code for each OS differently and didnt want to have dependencies. My plan was to use it PowerShell for .rar format as well but it didnt work out. So few months ago I implemented WinRAR, and after that I considered using it as main file extractor, but it was somehow slower compared to PowerShell so I didnt. 

I will add option to use WinRAR for all extraction when possible.

**Comment by m0lDaViA** [comment]
Is there a specific reason for ts? I'm not familiar with ts, that's why I'm asking^^

**Comment by Skitch90** [comment]
> Is there a specific reason for ts? I'm not familiar with ts, that's why I'm asking^^

do you mean the reason why it uses powershell? It's just a library (basically code written by another person)

**Comment by mienaiyami** [comment]
> Is there a specific reason for ts? I'm not familiar with ts, that's why I'm asking^^

By ts, do you mean typescript?

**Comment by m0lDaViA** [comment]
> > Is there a specific reason for ts? I'm not familiar with ts, that's why I'm asking^^
> 
> By ts, do you mean typescript?

Yes. Just curious^^

**Comment by mienaiyami** [comment]
If you mean typescript over javascript, then its because its a lot better compared to using javascript in almost everything.

If you are asking why typescript/javascript, then in short, its because Im using electronJS as backend and reactJS for the UI. It was a lot easier to make UI in react because I already knew js when I started building this app. 
Initially I just made a simple local webapp to read for myself, but due to filesystem limitations I turned to electron and made https://github.com/mienaiyami/offline-manga-reader. My knowledge was only limited to html, css and basic js at that time so this app was only usable. But with time it got complex for me with javascript because of errors and type, so I tried ts and implemented that in.
And I almost stopped working on it. And when i was good enough at Reactjs and typescript I tried using them to make a better version.

I tried doing it using other frameworks and languages but they didnt provide enough freedom for what I wanted. Some had high RAM usage while other had high idle CPU usage. 
And on top of that electron makes it easier to make apps for all OS while keeping most functionality same.
